tests/vanilla/memoryleaks.test.ts

```diff
@@ -1,14 +1,26 @@
 import LeakDetector from 'jest-leak-detector'
 import { expect, it } from 'vitest'
-import { atom, createStore } from 'jotai/vanilla'
+import { Atom, atom, createStore } from 'jotai/vanilla'
 
-it('should not have memory leaks with an atom', async () => {
+it('should not hold onto atoms that are not mounted', async () => {
   const store = createStore()
-  let detector: LeakDetector
-  ;(() => {
-    const objAtom = atom({})
-    detector = new LeakDetector(store.get(objAtom))
-  })()
```

Comment:
Hm, I somehow thought function closure would be necessary, but I think I got something wrong. Let me update my PR.